src/main/java/training/endava/playground/generics/types/LimitedContainer.java

```diff
@@ -0,0 +1,20 @@
+package training.endava.playground.generics.types;
+
+import java.util.ArrayList;
+
+public class LimitedContainer<T, N extends Number> {
+    ArrayList<T> container = new ArrayList<T>();
+    N limit;
+
+    public LimitedContainer(Class<T> c, N nr) {
+        this.limit = nr;
+    }
+
+    public boolean add(T obj) {
+        if (this.container.size() < (Integer) this.limit) {
```

Comment:
I don't think that's the way of comparing the filed of type T extends Number with an int. what happens if T is Double?
maybe smth like this: https://stackoverflow.com/questions/2683202/comparing-the-values-of-two-generic-numbers